go-controller/pkg/ovn/gateway/gateway.go

```diff
@@ -57,3 +64,45 @@ func GetGatewayPhysicalIPs(nbClient client.Client, gatewayRouter string) ([]stri
 
 	return nil, fmt.Errorf("no physical IPs found for gateway %s", gatewayRouter)
 }
+
+type MacBindingsMachine struct {
+	nodeLister corev1listers.NodeLister
+	sbClient   libovsdbclient.Client
+}
+
+func NewGatewayMacBindingMachine(nodeLister corev1listers.NodeLister, sbClient libovsdbclient.Client) *MacBindingsMachine {
+	return &MacBindingsMachine{
+		nodeLister: nodeLister,
+		sbClient:   sbClient,
+	}
+}
+
+func (mbm *MacBindingsMachine) BindOnNodes(datapathNamePrefix string) error {
+	nodes, err := mbm.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	dummyNextHopMAC := util.IPAddrToHWAddr(net.ParseIP(ovntypes.V4DummyNextHopMasqueradeIP))
+	for _, node := range nodes {
+		logicalPort := ovntypes.GWRouterToExtSwitchPrefix + node.GetName()
+		if err := util.CreateMACBinding(
+			mbm.sbClient,
+			logicalPort,
+			externalSwitchName(datapathNamePrefix, node.GetName()),
```

Comment:
the data path here is wrong it needs to be the GR

Reply:
Ah! I see. Good catch.